ultet/src/main/scala/za/co/absa/ultet/parsers/PgTableFileParser.scala

```diff
@@ -0,0 +1,122 @@
+package za.co.absa.ultet.parsers
+
+import cats.syntax.either._
+import io.circe.generic.auto._
+import io.circe.{yaml, _}
+import za.co.absa.ultet.dbitems.DBTable
+import za.co.absa.ultet.dbitems.DBTableMember.{DBTableColumn, DBTableIndex, DBTablePrimaryKey}
+import za.co.absa.ultet.model._
+
+import java.net.URI
+import java.nio.file.{Files, Paths}
+import java.util.stream.Collectors
+
+
+case class PgTableFileParser() {
+  import PgTableFileParser.DBTableFromYaml
+
+  def parseContentYaml(content: String): DBTableFromYaml = {
+    val loadedYaml = yaml.parser.parse(content)
+    val processedYaml = loadedYaml
+      .leftMap(err => err: Error)
+      .flatMap(_.as[DBTableFromYaml])
+      .valueOr(throw _)
+    processedYaml
+  }
+
+  def parseFileYaml(fileUri: URI): DBTableFromYaml = {
+    val path = Paths.get(fileUri)
+    val lines = Files.lines(path)
+    val content = lines.collect(Collectors.joining("\n"))
+
+    parseContentYaml(content)
+  }
+
+}
+
+
+object PgTableFileParser {
+
+  case class DBTableFromYaml(
+    tableName: String,
+    schemaName: String,
+    description: Option[String],
+    primaryDBName: String,
+    owner: String,
+    columns: Seq[Map[String, String]] = Seq.empty,
+    primaryKey: Option[Map[String, String]] = None,
+    indexes: Seq[Map[String, String]] = Seq.empty
+  ) {
+    private def prepareColumns: Seq[DBTableColumn] = {
+      columns.map(
+        currCol => {
+          DBTableColumn(
+            ColumnName(currCol("columnName")),
+            currCol("dataType"),
+            currCol("notNull").toBoolean,
+            currCol.get("description"),
+            currCol.get("default"),
+          )
+        }
+      )
+    }
+
+    private def prepareIndexes: Seq[DBTableIndex] = {
+      indexes.map(
+        currIndex => {
+          DBTableIndex(
+            currIndex("indexName"),
+            currIndex("tableName"),
+            currIndex("indexBy")
+              .replaceAll("^\\[|\\]$", "")
```

Comment:
```suggestion
              .replaceAll("""^\[|\]$""", "")
```
So you don't have to backslash the backslash

Reply:
sure, thanks for the tip